src/preact/base-element.js

```diff
@@ -329,51 +336,46 @@ export class PreactBaseElement extends AMP.BaseElement {
 
     this.renderDeferred_ = new Deferred();
     this.scheduleRender_();
+
+    if (Ctor['loadable']) {
+      this.setReadyState(ReadyState.LOADING);
+    }
+    this.maybeUpdateReadyState_();
+
     return this.renderDeferred_.promise;
   }
 
   /** @override */
-  layoutCallback() {
+  ensureLoaded() {
     const Ctor = this.constructor;
     if (!Ctor['loadable']) {
-      return super.layoutCallback();
+      return;
     }
-
     this.mutateProps(dict({'loading': Loading.EAGER}));
+    this.resetLoading_ = true;
```

Comment:
That's a good question and I've been wrestling with this a bit. Here's what happens: `ensureLoaded()` is called and it sets `loading ="eager"`. However, after that it stays "eager" and I thought that's illogical. Once the rendering happen with "eager", IMHO it should go back to "auto" because this is not a user-set state. Also it works together with `unlayoutOnPause`, which I really still hate but don't see a way around it. But the concept is this: we have to kill an iframe to pause, but we still need to be able to reload it when someone asks eager loading.

Reply:
What if the user actually does set `loading=eager` (or loading="XYZ") in the interim? Now we've clobbered their value.

Reply:
We currently do not allow setting `loading` value. But when we do then "eager" would have to reset to the value specified by the user instead of "auto", I'd think. In other words this "eager" value is transient when we do `ensureLoaded()`. However, the API doesn't exist yet, so I'm not sure yet how that will change.

Reply:
That makes me think `unlayoutOnPause` is more important than `eager`. When `!playable && unlayoutOnPause`, then `loading` should reset. Otherwise, setting `eager` should be persistent?

Reply:
Ok. So there are two things involved: `loading=eager` and `unloadOnPause`. I think we should treat them separately in this thread. This one is specific to the `ensureLoaded()` API which is meant to be transient: once the element is loaded it should no longer require eager loading for the future. Does this make sense? I think `unloadOnPause` is for now a bit confusing and we'll likely be able to improve on it later. But the transient nature of the `ensureLoaded` should be a lot clearer. LMK WYT.

Reply:
Ah. Is this only relevant in AMP/Bento Modes and not in Preact mode? In which case LGTM